src/libraries/Microsoft.PowerFx.Core/Utils/Extensions.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Globalization;
+using System.Net;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    internal static class Extensions
+    {
+        public static string GetDetailedExceptionMessage(this Exception ex, int maxDepth = 10)
+        {
+            return GetDetailedExceptionMessageInternal(string.Empty, ex, 0, maxDepth);
+        }
+
+        private static string GetDetailedExceptionMessageInternal(string prefix, Exception ex, int depth, int maxDepth)
+        {
+            depth++;
+
+            if (depth > maxDepth)
+            {
+                return string.Empty;
+            }
+
+            var sb = new StringBuilder();
```

Comment:
> var sb = new StringBuilder(); [](http://example.com/codeflow?start=11&length=30)

nit - A pattern here is to create the stringBuilder once on line 15 and pass it in.  That will play better with recursion. 

See  FormulaValue.ToExpression as an example.